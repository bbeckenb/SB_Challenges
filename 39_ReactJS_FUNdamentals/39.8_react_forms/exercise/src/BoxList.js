import React, { useState } from "react";
import "./BoxList.css"
import { v4 as uuid } from 'uuid';
import Box from "./Box";
import NewBoxForm from "./NewBoxForm";

function BoxList() {
    const [listOfBoxes, updateBoxes] = useState([]) //{id: uuid(), color:"aqua", width:"200", height:"200"}
    function removeBox(boxId) {
        updateBoxes(boxes => boxes.filter(b => b.id !== boxId))
    }
    function addBox(newBox) {
        updateBoxes(boxes => [...boxes, { ...newBox, id: uuid() }])
    }


    return (
        <div>
            <h1>Make Boxes!</h1>
            <NewBoxForm addBox={addBox} />
            <div>
            {listOfBoxes.map(({id, color, height, width}) => <Box key={id} id={id} color={color} width={width} height ={height} removeBox={evt => removeBox(`${id}`)}/>)}
            </div>
            
        </div>
    )
}

export default BoxList;