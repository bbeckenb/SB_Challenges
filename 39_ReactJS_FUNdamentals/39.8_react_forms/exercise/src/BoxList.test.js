import React from "react";
import {render, fireEvent} from "@testing-library/react";
import BoxList from "./BoxList";

it("renders without crashing", function() {
    render(<BoxList />);
  });
  
  it("matches snapshot", function() {
    const { asFragment } = render(<BoxList />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should add new box", function () {
    const { queryByText, getByLabelText } = render(<BoxList />);
    const input = getByLabelText("Color");
    const btn = queryByText("Add Box!");
    expect(queryByText('X')).not.toBeInTheDocument();
    fireEvent.change(input, { target: { value: '#FF0000' } });
    fireEvent.click(btn);
    expect(queryByText('X')).toBeInTheDocument();
  })

  it("should delete existing box", function () {
    const { queryByText } = render(<BoxList />);
    const btn = queryByText("Add Box!");
    expect(queryByText('X')).not.toBeInTheDocument();
    fireEvent.click(btn);
    expect(queryByText('X')).toBeInTheDocument();
    const delBtn = queryByText('X');
    fireEvent.click(delBtn);
    expect(queryByText('X')).not.toBeInTheDocument();
  })