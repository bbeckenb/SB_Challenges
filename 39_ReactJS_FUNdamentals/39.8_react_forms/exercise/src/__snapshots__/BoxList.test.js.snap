// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Make Boxes!
    </h1>
    <form>
      <label
        for="color"
      >
        Color
      </label>
      <input
        id="color"
        name="color"
        type="color"
        value="#000000"
      />
      <label
        for="height"
      >
        Height:75
      </label>
      <input
        id="height"
        max="200"
        min="50"
        name="height"
        type="range"
        value="75"
      />
      <label
        for="width"
      >
        Width:75
      </label>
      <input
        id="width"
        max="200"
        min="50"
        name="width"
        type="range"
        value="75"
      />
      <button>
        Add Box!
      </button>
    </form>
    <div />
  </div>
</DocumentFragment>
`;
